pandas/core/dtypes/missing.py

```diff
@@ -188,6 +191,12 @@ def _use_inf_as_na(key):
     """
     inf_as_na = get_option(key)
     globals()["_isna"] = partial(_isna, inf_as_na=inf_as_na)
+    if inf_as_na:
+        globals()["nan_checker"] = lambda x: ~np.isfinite(x)
```

Comment:
don't do it like this

use the global keyword

Reply:
this is the pattern we already use a few lines up

Reply:
really? ok for here, but we *really* should just use the `global` keyword as this is the purpose.